controllers/rhmi/rhmi_controller.go

```diff
@@ -1367,6 +1384,108 @@ func (r *RHMIReconciler) createInstallationCR(ctx context.Context, serverClient
 	return installation, nil
 }
 
+func (r *RHMIReconciler) composeAndSetAlertMetric(installation *rhmiv1alpha1.RHMI, rc *rest.Config, configManager *config.Manager) error {
+	var alerts []resources.AlertMetric
+
+	alertingNamespaces, err := r.getAlertingNamespace(installation, configManager)
+	if err != nil {
+		return fmt.Errorf("getting alerting namespace failed: %w", err)
+	}
+
+	observability, err := configManager.ReadObservability()
+	if err != nil {
+		return fmt.Errorf("getting observability configuration failed: %w", err)
+	}
+
+	for namespace, _ := range alertingNamespaces {
+		if namespace == observability.GetNamespace() {
+			alerts, err = r.composeAlertMetric("prometheus", namespace, rc)
+			if err != nil {
+				return fmt.Errorf("composing alert metric failed: %w", err)
+			}
+			metrics.SetRHOAMAlerts(alerts)
+		}
+	}
+
+	return nil
+}
+
+func (r *RHMIReconciler) composeAlertMetric(route string, namespace string, rc *rest.Config) ([]resources.AlertMetric, error) {
+	var alerts resources.AlertMetrics
+	endpoint := "/api/v1/alerts"
+
+	url, err := r.getURLFromRoute(route, namespace, rc)
+	if err != nil {
+		return alerts.Alerts, fmt.Errorf("error getting route : %w", err)
+	}
+	url = url + endpoint
+
+	req, err := http.NewRequest("GET", url, nil)
```

Comment:
It just feels neater to me. I don't think it could happen because of how reconcilers work but the `url = url + endpoint` could end up with the url and endpoint getting added again and again.